### PR TITLE
Let's Encrypt Caddybox

### PR DESCRIPTION
**Pull request by joshix**
Towards #14 and friends.

Make it easy to TLS sites served with Caddybox.
Requires a resolving DNS name pointing to the container's listen.

Make caddy run as root, so can bind 80 and 443.
Add root Certificate Authorities file from CoreOS v891.0.0.
Add a bit of usage doc to README.

### CODE REVIEW

**Comment by cboettig** [comment]
Looks good to me.  

Very minor: In the markdown documentation, can you break the `docker run` command over a few lines so it doesn't run off the page?

 Question: Why not link a second volume to `/.caddy/letsencrypt` when you first run?  Then presumably you wouldn't need `docker cp` and you would always use the same `docker run` command.

**Comment by cboettig** [comment]
Hmm, just gave this a whirl, looks like certificates are being created correctly (I link `-v $HOME/.caddy/letsencrypt:/.caddy/letsencrypt`), log looks correct, no errors, and yet when I try and connect to the site I simply get an error site not found.  If I shut down the container and just run the native `caddy` binary in the same container, everything works fine (using the credentials that `caddybox` created in the shared volume; once I change them back to user-owned permissions).  Any ideas?

**Comment by joshix** [comment]
@cboettig Are you building the container from the sources here, or pulling it from docker reg? If you're pulling from docker reg, you need to specify the `joshix/caddy:le` tag. If you're building, you must build from this branch (also called `le`). Those are my first ideas without having repeated your problem myself.

**Comment by cboettig** [comment]
Yup, I tried both building from the le branch and simply using `joshix/caddy:le`

**Comment by joshix** [comment]
Hm. Wanna post your `docker run` invocation and we'll see if we can figure this out?

Is the specific error a `404 not found` shown in the browser from caddy itself, or what's the output?

**Comment by cboettig** [comment]
Ah, got it, I somehow typed `433:433` instead of `443:443`, silly me.  All working now.

**Comment by joshix** [comment]
\o/

File: Dockerfile [hunk 1]
```diff
@@ -2,6 +2,5 @@ FROM scratch
 MAINTAINER Josh Wood <j@joshix.com>
 COPY rootfs /
 EXPOSE 2015
-USER  22015
```
Comment (joshix) [inline Dockerfile]: Boo & Hiss to returning to EUID 0

**Pull request merged**